sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -460,39 +460,43 @@ public String getId() {
         });
     }
 
-    private <T> Flux<FeedResponse<T>> applyTransformer(Flux<FeedResponse<Document>> queryDocuments, Function<Document, Document> transformer, Scheduler scheduler, Class<T> classType) {
+    private <T> Flux<FeedResponse<T>> applyTransformer(Flux<FeedResponse<Document>> queryDocuments, Function<Document, Mono<Document>> transformer, Scheduler scheduler, Class<T> classType) {
         if (transformer == null) {
-            return queryDocuments.map(response -> prepareFeedResponse(response, classType));
+            return queryDocuments.flatMap(response -> prepareFeedResponse(response, classType));
         } else {
-            return queryDocuments.publishOn(scheduler).subscribeOn(scheduler).map(response -> prepareFeedResponse(response, classType, transformer));
+            return queryDocuments.publishOn(scheduler).subscribeOn(scheduler).flatMap(response -> prepareFeedResponse(response, classType, transformer));
         }
     }
 
-    private <T> FeedResponse<T> prepareFeedResponse(FeedResponse<Document> response, Class<T> classType) {
+    private <T> Mono<FeedResponse<T>> prepareFeedResponse(FeedResponse<Document> response, Class<T> classType) {
         return prepareFeedResponse(response, classType, null);
     }
 
-    private <T> FeedResponse<T> prepareFeedResponse(FeedResponse<Document> response, Class<T> classType, Function<Document, Document> transformer) {
+    private <T> Mono<FeedResponse<T>> prepareFeedResponse(FeedResponse<Document> response, Class<T> classType, Function<Document, Mono<Document>> transformer) {
         QueryInfo queryInfo = ModelBridgeInternal.getQueryInfoFromFeedResponse(response);
         if (queryInfo != null && queryInfo.hasSelectValue()) {
-            List<T> transformedResults = response.getResults()
+            List<Mono<T>> transformedResultsAsync = response.getResults()
                                              .stream()
                                               .map(d -> d.has(Constants.Properties.VALUE) ?
-                                                 transform(d.get(Constants.Properties.VALUE), classType) :
+                                                 Mono.just(transform(d.get(Constants.Properties.VALUE), classType)) :
                                                  ModelBridgeInternal.toObjectFromJsonSerializable(d, classType, transformer))
                                              .collect(Collectors.toList());
 
-            return BridgeInternal.createFeedResponseWithQueryMetrics(transformedResults,
-                                                                     response.getResponseHeaders(),
-                                                                     ModelBridgeInternal.queryMetrics(response));
-
+            return Flux.merge(transformedResultsAsync).collectList().map(
```

Comment:
Regarding these two changes (this and below) with using `Flux.merge` - did you do any performance testing?
Can you please share the results ?
We should make sure this doesn't break any existing cosmos APIs. Please run the full CI for this PR.

Reply:
all the enrcyption code for query factored out.
This has 0 impact on perf now.